specs/experimental/security-council-vetoerset.md

```diff
@@ -0,0 +1,206 @@
+1. The Safe Account cannot have more owners than the configured `maxOwnerCount`.
+2. The Safe Account `threshold` (minimum number of signatures necessary to perform an arbitrary action on behalf of the Safe Account) must always be exactly equal to 66% of the current number of owners.
+
+This guard has the following minimal interface:
+
+```solidity
+interface VetoModule {
+    /// @notice Inherited hook from `BaseGuard` that is run right before the transaction is executed
+    ///         by the Safe Account when `execTransaction` is called.
+    function checkTransaction(
+        address,
+        uint256,
+        bytes memory,
+        Enum.Operation,
+        uint256,
+        uint256,
+        uint256,
+        address,
+        address payable,
+        bytes memory,
+        address
+    )
+        external;
+
+    /// @notice Inherited hook from `BaseGuard` that is run right after the transaction has been executed
+    ///         by the Safe Account when `execTransaction` is called.
+    function checkAfterExecution(bytes32, bool) external view;
+
+    /// @notice Update the maximum number of owners.
+    /// @dev Reverts if not called by the Safe Account.
+    /// @param newMaxOwnerCount The new possible `maxOwnerCount` of owners.
+    function updateMaxOwnerCount(uint8 newMaxOwnerCount) external;
+
+    /// @notice Checks if the given `newOwnerCount` of owners is allowed and returns the corresponding 66% threshold.
+    /// @dev Reverts if `newOwnerCount` is above `maxOwnerCount`.
+    /// @param newOwnerCount The owners count to check.
+    /// @return threshold The corresponding 66% threshold for `newOwnerCount` owners.
+    function checkNewOwnerCount(uint256 newOwnerCount) public view returns (uint256 threshold);
+}
+```
+
+Note that the `checkTransaction()` and `checkAfterExecution()` hooks are called [before](https://github.com/safe-global/safe-smart-account/blob/main/contracts/Safe.sol#L137-L156) and [after](https://github.com/safe-global/safe-smart-account/blob/main/contracts/Safe.sol#L179-L183) executing transactions on the Safe Account.
+
+The current implementation keeps `checkTransaction()` empty and enforces both conditions described earlier in the `checkAfterExecution()` post hook.
+
+`updateMaxOwnerCount()` allows a signature quorum to update the `maxOwnerCount` limit while always ensuring it stays above or equal to the current number of owners of the Safe Account.
+
+### Adding owner call flow
+
+The following diagram illustrates the flow for adding a single owner using the `AddOwnerModule`'s `addOwner()` method:
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant AddOwnerModule
+    participant OwnerGuard
+    participant Safe
+    
+
+    User->>AddOwnerModule: addOwner([owner])
+    activate AddOwnerModule
+        Note over AddOwnerModule: reverts if sender is not the OP Foundation
+
+        AddOwnerModule->>OwnerGuard: checkNewOwnerCount([currentOwnerCount] + 1)
```

Comment:
It feels odd to me that the `AddOwnerModule` calls directly to the `OwnerGuard`. I think you can achieve the same effect by doing the max owner check in `checkAfterExecution`.

Reply:
Yeah it's a bit a bummer the Safe Account version used (1.4.0) did not run the guard when t[he transaction is performed from a module](https://github.com/safe-global/safe-smart-account/blob/v1.4.0/contracts/base/ModuleManager.sol#L90). The very last version does something different but we haven't adopted it yet.